workspace/apps/pidp/src/app/features/access/pages/bc-provider-application/bc-provider-application.component.ts

```diff
@@ -99,14 +104,17 @@ export class BcProviderApplicationComponent
   protected performSubmission(): Observable<string | void> {
     const partyId = this.partyService.partyId;
     this.password = this.formState.password.value;
+    this.loadingOverlayService.open(LOADING_OVERLAY_DEFAULT_MESSAGE);
 
     return this.resource.createBcProviderAccount(partyId, this.password).pipe(
       map((upn: string) => {
         this.username = upn;
         this.completed = true;
+        this.loadingOverlayService.close();
         this.showSuccessDialog();
       }),
       catchError(() => {
+        this.loadingOverlayService.close();
```

Comment:
Maybe we should use the [`finalize`](https://www.learnrxjs.io/learn-rxjs/operators/utility/finalize) RxJS operator rather than copy/paste the same code in a success and error case

Reply:
tried but it closes the success dialog as well